paper_sections/03-hierarchical_gams.Rmd

```diff
@@ -356,5 +357,40 @@ bird_mod5 <- gam(count ~ te(week, latitude, by=species, bs= c("cc", "ts"),
                  data=bird_move, method="REML", family=poisson)
 ```
 
-**GLS: this seems a little outof place here? Needs a better segue into it is it stays here and we need a model/formula example!** Where group-level smooths are coded using the `by` argument if the factor is unordered, *mgcv* will set up a model with one smooth for each grouping level. If the factor is ordered, *mgcv* will not set the basis functions for the first grouping level to zero. In model 3 (with an ungrouped smooth included) the ungrouped smooth will then correspond to the first grouping level, rather than the average functional response, and the group-specific smooths will correspond to deviations from the first group. In model 5, using an ordered factor will result in the first group not having a smooth term associated with it at all.
+
+## Comparing different HGAM specifications
+
+These models can be compared using standard model comparison tools. Model 2 and model 3 will generally be nested in model 1 (depending on how each model is specified) so ANOVA comparisons may be used to test if the groupwise smoother is necessary. However, we do not currently recommend this method. Given the uncertainty about what degrees of freedom to assign to models with varying smooths, and the fact that slightly different model specifications may not result in nested models, we do not think there is sufficient theory on how accurate parametric p-values will be for comparing these models (see `?mgcv::anova.gam` for more discussion on ANOVA comparisons for GAMs).  
+
+Comparing models based on AIC is a more robust approach to comparing the different modelling approaches, as there is a well-developed theory of how to include smoothing effects when estimating the model complexity penalty for AIC [@wood_smoothing_2016]. We demonstrate this approach in Table \ref{tab:AIC_table}. Going by the AIC criteria, there is strong support for including among-group functional variability for both the CO2 dataset and the bird_move dataset (compare models 1 versus models 2-5). For the CO2 dataset (Table \ref{tab:AIC_table}A), there is relatively strong evidence that there is more intergroup variability in smoothness than model 2 allows, and weaker evidence that model 4 or 5 (separate smooths for all plants) show the best fit. For the bird_move dataset (Table \ref{tab:AIC_table}B), model 2 (global smooth plus group-level smooths with a shared penalty) fits the data best, which is good as that is how we simulated the data! We have also included the adjusted $R^2$ for all models in Table \ref{tab:AIC_table}, but we do not recommend selecting a model based on this, for the same reasons that we do not recommend parametric p-values. 
```

Comment:
Okay, so I think we should drop the R^2 stuff here.

We could talk about predictive checks too? (obs vs. expected type stuff?)